src/frontend/src/optimizer/rule/agg_on_index_rule.rs

```diff
@@ -0,0 +1,60 @@
+use super::{BoxedRule, Rule};
+use crate::optimizer::plan_node::{LogicalLimit, LogicalScan, LogicalTopN};
+use crate::optimizer::property::Direction::Asc;
+use crate::optimizer::property::{FieldOrder, Order};
+use crate::optimizer::PlanRef;
+
+pub struct AggOnIndexRule {}
+
+impl Rule for AggOnIndexRule {
+    fn apply(&self, plan: PlanRef) -> Option<PlanRef> {
+        let logical_topn: &LogicalTopN = plan.as_logical_top_n()?;
+        let logical_scan: LogicalScan = logical_topn.get_child().as_logical_scan()?.to_owned();
```

Comment:
This rule name is inconsistent with its match pattern. Maybe `TopNOnIndexRule` is more appropriate.